include/deal.II/lac/affine_constraints.templates.h

```diff
@@ -2610,11 +2610,18 @@ AffineConstraints<number>::distribute(VectorType &vec) const
       // following.
       IndexSet needed_elements = vec_owned_elements;
 
+      std::vector<types::global_dof_index> additional_elements;
       for (const ConstraintLine &line : lines)
         if (vec_owned_elements.is_element(line.index))
           for (const std::pair<size_type, number> &entry : line.entries)
-            if (!vec_owned_elements.is_element(entry.first))
-              needed_elements.add_index(entry.first);
+            additional_elements.emplace_back(entry.first);
```

Comment:
Can you please keep the `if` statement? Otherwise, the vector is going to be really big, whereas it should be enough to set up the entries that are not already in `vec_owned_elements`/`needed_elements`.

Reply:
Sure thing!